test/pairwise_fail_test.c

```diff
@@ -19,6 +19,7 @@ typedef enum OPTION_choice {
     OPT_CONFIG_FILE,
     OPT_PAIRWISETEST,
     OPT_DSAPARAM,
+    OPT_FIPSVERSION,
```

Comment:
Elsewhere we do the version test in the .t file and skip not the C.  Any reason for doing it this way instead?

Reply:
Can you point me to an example in a .t?

Reply:
It's something like this in the SKIP section:
```perl
    run(test(["fips_version_test", "-config", $provconf, ">=3.1.0"]),
             capture => 1, statusvar => \my $exit);
    skip "FIPS provider version too old", 1
        if !$exit;
```
A bit of setup is required: https://github.com/openssl/openssl/blob/e788c772b12eea5ced4ce46619e13acf0e0eb6ba/test/recipes/25-test_verify.t#L355-L367

Fortunately, the setup and check only needs to be done once and can be used in both places you require it.